docs/pages/management/dynamic-configuration/spacelift.mdx

```diff
@@ -0,0 +1,298 @@
+**CONTINUE**. Do the same with the **DEFINE BEHAVIOR** tab and click **SAVE
+STACK**.
+
+Your new stack should resemble the following:
+
+![New Spacelift stack](../../../img/management/spacelift/newstack.png)
+
+## Step 3/4. Grant Teleport permissions to Spacelift
+
+In this section, you will create a Teleport user and role for Spacelift, plus a
+role that can *impersonate* the Spacelift user in order to export an identity
+file. You will then export an identity file and upload it to Spacelift.
+
+Since Teleport manages RBAC permissions via configuration resources, and you
+have not set up Spacelift yet, you must create the resources in this section
+outside of Spacelift. After that, Spacelift can manage all of your Teleport
+configuration resources.
+
+### Create a Spacelift user and role
+
+Create a local Teleport user named `spacelift` and a matching role granting the
+necessary permissions for Terraform to manage resources in your cluster.
+
+On your workstation, _outside_ the git repository you connected to Spacelift,
+add the following content to a file called `spacelift.yaml`:
+
+```yaml
+kind: role
+metadata:
+  name: spacelift
+spec:
+  allow:
+    rules:
+      - resources:
+        - role
+        - user
+        verbs: ['list','create','read','update','delete']
+version: v6
+---
+kind: user
+metadata:
+  name: spacelift
+spec:
+  roles: ['spacelift']
+version: v2
+```
+
+This is a minimal version of the configuration you will need to provide to
+Spacelift, and is only permitted to manage Teleport roles. 
+
+Create the `spacelift` user and role.
+
+```code
+$ tctl create spacelift.yaml
+role 'spacelift' has been created
+user "spacelift" has been created
+```
+
+### Enable impersonation
+
+The `spacelift` user cannot log in to Teleport to retreive credentials, so
+another user must **impersonate** this user in order to request credentials on
+`spacelift`'s behalf.
+
+Create a role that enables your user to impersonate the Terraform user. Paste
+the following YAML document into a file called `spacelift-impersonator.yaml`:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: spacelift-impersonator
+spec:
+  allow:
+    impersonate:
+      users: ['spacelift']
+      roles: ['spacelift']
+```
+
+Next, create the role:
+
+```code
+$ tctl create spacelift-impersonator.yaml
+```
+
+(!docs/pages/includes/add-role-to-user.mdx role="spacelift-impersonator"!)
+
+### Export an identity file
+
+(!docs/pages/includes/plugins/identity-export.mdx user="spacelift" client="Spacelift"!)
+
+### Provide an identity file to Spacelift
+
+Click **Stacks > Teleport**. Click the **Environment** tab, then **Edit**. Set
+the dropdown menu that follows to **Mounted file**, and the path to
+`/mnt/workspace/auth.pem`.
+
+Click **Upload file** and select the file called `auth.pem` that you exported
+earlier.
+
+Make sure you click **Secret** next to the entry for the mounted file:
+
+![Uploading the identity file](../../../img/management/spacelift/id-file.png)
```

Comment:
If the purpose of this image is to point out the **Secret** option, do we want to highlight it, perhaps something like this?

![image](https://github.com/gravitational/teleport/assets/2349184/5ce03269-7b6e-4a9f-8009-b49d4b42e5e7)

Reply:
I wanted to show the overall view. Does it look like readers will miss the "Secret" option if we don't highlight it?

Reply:
If the point of the image is to show the user where the Secret option is, I would. If the point of the image is to show what the end state should look like, I would end the statement above with a period and add a new statement to the affect of "The Environment tab should look like this".